com.unity.mobile.notifications/Runtime/iOS/iOSNotificationsWrapper.cs

```diff
@@ -104,13 +102,13 @@ public static void RegisterOnReceivedCallback()
         }
 
         [MonoPInvokeCallback(typeof(AuthorizationRequestCallback))]
-        public static void AuthorizationRequestReceived(IntPtr authRequestDataPtr)
+        public static void AuthorizationRequestReceived(IntPtr request, IntPtr authRequestDataPtr)
```

Comment:
Since you're changing this code, can you make it even more safer. Add the incoming iOSAuthorizationRequestData  struct size, so you can then verify that incoming struct size matches struct size on C#, if it doesn't match throw an excption.

Also, I never understood why do we need Marshal.PtrToStructure ? and why not pass struct type.

So in short:
1. Change these types
private delegate void AuthorizationRequestCallback(IntPtr request, int size, iOSAuthorizationRequestData data);
public static void AuthorizationRequestReceived(IntPtr request,int size, iOSAuthorizationRequestData data)
2. Add validation for incoming objc iOSAuthorizationRequestData  size versus C# iOSAuthorizationRequestData size, if doesn't match throw an error, this will help avoid case where somebody updates or incorrectly adds type to struct
3. Remove Marshal.PtrToStructure and just use iOSAuthorizationRequestData directly, unless you know a reason why we should keep Marshal.PtrToStructure here

Reply:
Same for  private delegate void NotificationReceivedCallback(IntPtr notificationData); 

1. Add int size parameter before notificationData, so you can perform validation
2. And maybe instead IntPtr use iOSNotificationData directly, since the struct layout matches layout from objc.

Reply:
P.S if these changes will be done, please rerun QA pass, just in case

Reply:
still waiting for size parameter for validation, just in case

Reply:
I don't like the idea of putting unrelated changes to this PR, so I'd like to resolve iOSNotificationData part in a separate PR.

As for iOSAuthorizationRequestData I took a bit different approach: a separate code that only runs once in development build. Lets not pollute production code with checks that must not fail.

Reply:
Can you log iOSNotificationData task somewhere so it wouldn't be forgotten?

Reply:
Working on it right away. But the thing is quite dreaded, it was a good idea to do it in separate PR.

Reply:
Thanks, i'll approve this PR then